src/prof_recent.c

```diff
@@ -495,7 +495,7 @@ prof_recent_alloc_dump_node(emitter_t *emitter, prof_recent_t *node) {
 	    &node->alloc_tctx->thr_uid);
 	prof_tdata_t *alloc_tdata = node->alloc_tctx->tdata;
 	assert(alloc_tdata != NULL);
-	if (alloc_tdata->thread_name != NULL) {
+	if (!prof_thread_name_empty(alloc_tdata)) {
 		emitter_json_kv(emitter, "alloc_thread_name",
 		    emitter_type_string, &alloc_tdata->thread_name);
```

Comment:
We are seeing a crash in `jemalloc_je_ctl_byname` -> `experimental_prof_recent_alloc_dump_ctl` -> `jemalloc_je_prof_recent_alloc_dump` as (I believe) a result of this change.

The problem is that `emitter_json_kv` -> `emitter_json_value` -> `emitter_print_value` [expects `emitter_type_string` parameters to be passed as a double pointer](https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/emitter.h#L165-L167), but because we changed `thread_name` from a `char *thread_name` to `char thread_name[]`, `&thread_name` is no longer `char **` -- it's just `char *`.

I think you might need to do something like this instead:

```
if (!prof_thread_name_empty()) {
  const char *thread_name = &alloc_tdata->thread_name;
  emitter_json_kv(emitter, "alloc_thread_name",
    emitter_type_string, &thread_name);
}
```